core/test/network-records-to-devtools-log.js

```diff
@@ -339,7 +339,8 @@ function getLoadingFailedEvent(networkRecord, index, normalizedTiming) {
     method: 'Network.loadingFailed',
     params: {
       requestId: getBaseRequestId(networkRecord) || `${idBase}.${index}`,
-      timestamp: normalizedTiming.networkEndTime,
+      timestamp: normalizedTiming.networkEndTime / 1000,
+      type: networkRecord.resourceType || undefined,
```

Comment:
Apparently we didn't have any `failed` requests going through `networkRecordsToDevtoolsLog` (at least with verification on) because the missing `/ 1000` would have caused any to fail the roundtrip test. Fixed the timing, added support for `failed` resourceTypes for good measure (needed for the document redirect chain debugging I was doing), and added the line to `network-records-to-devtools-log-test.js` to ensure that path is exercised going forwards